Sources/NIOIMAPCore/Grammar/Message/MessageAttributes.swift

```diff
@@ -58,6 +58,12 @@ public enum MessageAttribute: Hashable {
 
     /// `X-GM-LABELS`: provides the labels for a given message
     case gmailLabels([GmailLabel])
+    
+    /// `PREVIEW`: a server-generated abbreviated text representation of message data that
+    /// is useful as a contextual preview of the entire message. 
+    ///
+    /// - SeeAlso: RFC 8970 (IMAP4 Extension: Message Preview Generation)
+    case preview(PreviewText)
```

Comment:
If the server responds something like
```
* 1 FETCH (UID 5647 PREVIEW NIL)
```
this should end up being `.preview(nil)` and as such this needs to be `Optional`:
```suggestion
    case preview(PreviewText?)
```

To do this `writeMessageAttribute_preview()` and `parseMessageAttribute_preview()` need to be updated. See below.